src/main/java/bitflow4j/steps/BatchPipelineStep.java

```diff
@@ -189,7 +189,14 @@ private void flush(List<Sample> window) throws IOException {
     // Printing ========================================
     // =================================================
     public String toString() {
-        return String.format("Batch processing, %s handler(s)", handlers.size());
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < handlers.size(); i++) {
+            sb.append(handlers.get(0).getClass().getSimpleName());
```

Comment:
I think this should be `handlers.get(i)`...
You could also simplify the loop to `for (Class cls : handlers)` to avoid this mistake.